pkg/lib/docker/docker.go

```diff
@@ -161,12 +161,20 @@ func PullImage(image string, encodedAuthConfig string, pullVerbosity PullVerbosi
 		jsonmessage.DisplayJSONMessagesStream(pullOutput, os.Stderr, termFd, isTerm, nil)
 		fmt.Println()
 	case PrintDots:
+		var err error
 		fmt.Printf("￮ downloading docker image %s ", image)
-		defer fmt.Print(" ✓\n")
+		defer func() {
+			if err == nil {
+				fmt.Print(" ✓\n")
+			} else {
+				fmt.Print(" x\n")
+			}
+		}()
 		dotCron := cron.Run(print.Dot, nil, 2*time.Second)
 		defer dotCron.Cancel()
 		// wait until the pull has completed
-		if _, err := ioutil.ReadAll(pullOutput); err != nil {
+		_, err = ioutil.ReadAll(pullOutput)
```

Comment:
@zouyee It looks like an error will only be thrown if the io reading fails. I could be wrong but it doesn't look like `ioutil.ReadAll(pullOutput)` would fail if the actual docker image pull fails.

Have you tried to reproduce the issue and test this solution? If you have, it would be great if you can share the error message. Ideally, the error message from docker should be displayed if the error message is easy to understand for the user. Otherwise, the error should be detected and a more understandable error should be thrown.

I think you may have to parse output from `pullOutput`, here is an example: https://stackoverflow.com/questions/44452679/golang-docker-api-parse-result-of-imagepull (instead of creating your own event struct you can use [jsonmessage.JSONMessage](https://godoc.org/github.com/docker/docker/pkg/jsonmessage).) Once the JSONMessages have been parsed, you can iterate through the messages and look for any JSONMessages with errors. From those errors, we can come up with a meaningful error message to expose to the user.